### PR TITLE
Add support for distributed RST files

### PR DESCRIPTION
**Pull request by greschd**
Use a list of paths instead of a single path in the `rst` member
of `ContinuousFiberCompositesFiles`, to allow passing multiple
RST files.
Add a constructor that can take either a single path or a list of
paths, and converts it to a list if needed.

Rename the `rst_file` argument to `rst_files` in the
`ResultDefinition` class, and convert it to a list.

Adapt the following functions to handle multiple RST files:

* `get_composite_data_sources`
* `upload_continuous_fiber_composite_files_to_server`
* `get_composite_files_from_workbench_result_folder`
* `get_continuous_fiber_example_files`

Add a `material_support` member to 'CompositeDataSources', since
DPF core cannot (currently) use a distributed DataSources object for
creating the material support. Adapt the `get_composite_data_sources`
function accordingly. This is a temporary workaround, until DPF core
supports creating a material support from a distributed DataSources.

The `ElementInfo` also currently fails, due to a bug in the mesh property
provider operator. The corresponding tests are marked as `xfail`. The
missing support is documented in the `ElementInfoProvider` docstring, 
and the resulting exception message.

### CODE REVIEW

File: src/ansys/dpf/composites/layup_info/_layup_info.py [hunk 1]
```diff
@@ -424,12 +429,14 @@ def get_element_info_provider(
 
     def get_keyopt_property_field(keyopt: int) -> PropertyField:
         keyopt_provider = dpf.Operator("mesh_property_provider")
+        # keyopt_provider = dpf.Operator("property_field_provider_by_name")
```
Comment (roosre) [inline src/ansys/dpf/composites/layup_info/_layup_info.py:432]: Is this change on purpose? And what about the change on line 439?

File: src/ansys/dpf/composites/data_sources.py [hunk 2]
```diff
@@ -38,13 +39,43 @@ class CompositeDefinitionFiles:
 class ContinuousFiberCompositesFiles:
     """Provides the container for continuous fiber composite file paths."""
 
-    rst: _PATH
+    rst: List[_PATH]
```
Comment (roosre) [inline src/ansys/dpf/composites/data_sources.py:42]: Can we rename this parameter to rsts or result_files?
rst is confusing because it is a list. And we plan to support LS Dyna or thermal analyses (rth) as well.

File: src/ansys/dpf/composites/data_sources.py [hunk 3]
```diff
@@ -67,6 +98,12 @@ class CompositeDataSources:
     """Provides data sources related to the composite lay-up."""
 
     rst: DataSources
```
Comment (roosre) [inline src/ansys/dpf/composites/data_sources.py:100]: Shouldn't that become a list as well?

File: src/ansys/dpf/composites/data_sources.py [hunk 4]
```diff
@@ -38,13 +39,43 @@ class CompositeDefinitionFiles:
...
+        if isinstance(rst, (str, pathlib.Path)):
+            rst = [rst]
+        self.rst = rst  # type: ignore
+        self.composite = composite
+        self.engineering_data = engineering_data
+        self.files_are_local = files_are_local
+
 
 @dataclass
 class ShortFiberCompositesFiles:
```
Comment (roosre) [inline src/ansys/dpf/composites/data_sources.py:81]: Should this data class be updated as well?

**Review by roosre — CHANGES_REQUESTED** [review]

File: src/ansys/dpf/composites/layup_info/_layup_info.py (hunk 1 above)
Comment (greschd) [reply src/ansys/dpf/composites/layup_info/_layup_info.py]: No, that was an unintentional leftover from testing. Reverted.

File: src/ansys/dpf/composites/data_sources.py (hunk 3 above)
Comment (greschd) [reply src/ansys/dpf/composites/data_sources.py:100]: No, a single `DataSources` object refers to all distributed RST files (added via either `set_result_file_path` for a single file, or `set_domain_result_file_path` for multiple RST).

File: src/ansys/dpf/composites/data_sources.py (hunk 4 above)
Comment (greschd) [reply src/ansys/dpf/composites/data_sources.py:81]: Hm, I haven't tested distributed RST in the short fiber context at all yet (neither here, nor in the back-end). Could still do this of course, but then we may as well create a separate PR.

File: src/ansys/dpf/composites/data_sources.py (hunk 2 above)
Comment (greschd) [reply src/ansys/dpf/composites/data_sources.py:42]: Hmm, to my eye `rst` is consistent with the `composite` attribute that also refers to multiple composite definition files.

For LS-Dyna and RTH files: Do you know if we'll need to explicitly distinguish between them, or can just use the `DataSources` interface on these to abstract away the differences? If we need to treat them explicitly, they probably need a separate attribute.

File: src/ansys/dpf/composites/data_sources.py (hunk 4 above)
Comment (roosre) [reply src/ansys/dpf/composites/data_sources.py:81]: OK. We can inform Federico that distributed RST are supported now and that he can test it for his operators.

**Review by roosre — APPROVED** [review]
